core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -328,6 +330,7 @@ private[spark] class MemoryStore(blockManager: BlockManager, maxMemory: Long)
     val droppedBlocks = new ArrayBuffer[(BlockId, BlockStatus)]
 
     accountingLock.synchronized {
+      releasePendingUnrollMemoryForThisThread()
```

Comment:
as commented in the main thread, I would move this to the end of this `accountingLock.synchronized` block and add a comment there:
```
// Release the memory used to unroll this block
// because we no longer need the underlying Array
releasePendingUnrollMemoryForThisThread()
```